rslib/src/scheduler/fsrs/simulator.rs

```diff
@@ -61,3 +74,20 @@ impl Collection {
         })
     }
 }
+
+impl Card {
+    fn convert(card: Card, col: &mut Collection) -> fsrs::Card {
+        if let Some(state) = card.memory_state {
+            let due = card.original_or_current_due();
+            let relative_due = due - col.timing_today().unwrap().days_elapsed as i32;
+            fsrs::Card {
+                difficulty: state.difficulty as f64,
+                stability: state.stability as f64,
+                last_date: (relative_due - card.interval as i32) as f64,
+                due: relative_due as f64,
+            }
+        } else {
+            todo!();
```

Comment:
What should happen if the card doesn't have memory state?

Reply:
It's OK to ignore them.

Reply:
todo!() will crash, so you'd need to return Some/None and filter out the Nones in the map().collect() call.

Reply:
The latest commit fixed it.